placeholders_test.go

```diff
@@ -0,0 +1,132 @@
+package txtdirect
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestParsePlaceholders(t *testing.T) {
+	tests := []struct {
+		url       string
+		requested string
+		expected  string
+	}{
+		{
+			"example.com{uri}",
+			"https://example.com/?test=test",
+			"example.com/?test=test",
+		},
+		{
+			"example.com{uri}/{~test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{uri}/{>Test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test-header",
+		},
+		{
+			"example.com{uri}/{?test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{dir}",
+			"https://example.com/directory/test",
+			"example.com/directory/",
+		},
+		{
+			"example.com/directory/{file}",
+			"https://example.com/directory/file.pdf",
+			"example.com/directory/file.pdf",
+		},
+		{
+			"example.com/?test={fragment}",
+			"https://example.com/?test=#fragment",
+			"example.com/?test=fragment",
+		},
+		{
+			"example.com/{host}",
+			"https://project.example.com:8080",
+			"example.com/project.example.com:8080",
+		},
+		{
+			"example.com/{hostonly}",
+			"https://project.example.com",
+			"example.com/project.example.com",
+		},
+		{
+			"example.com/{method}",
+			"https://example.com",
+			"example.com/GET",
+		},
+		{
+			"example.com{path}",
+			"https://example.com/path",
+			"example.com/path",
+		},
+		{
+			"example.com{path_escaped}",
+			"https://example.com/path_escaped",
+			"example.com%2Fpath_escaped",
+		},
+		{
+			"example.com:{port}",
+			"https://example.com:8080",
+			"example.com:8080",
+		},
+		{
+			"subdomain.example.com/{label1}",
+			"https://subdomain.example.com/subdomain",
```

Comment:
Using "subdomain" also for the uri or path within the request makes the test harder to understand.
Same for the `label` tests below.

Reply:
Sounds good - I changed the {labelN} tests so that they're easier to understand.